### DISCUSSION TITLE
Grid action strip position

### DISCUSSION THREAD

**Discussion by Eralmidia** [IgniteUI/igniteui-angular#10624] [Q&A]
We are looking at utilizing the action strip for some of our grids. Are there any options of controlling the positioning of the strip? Looking at the sample at https://www.infragistics.com/angular-demos/grid/grid-add-row the actions are positioned to the right side. Is there a way to instead lock the position to a specific column? We feel that on a wide grid/large screen, the actions could be easily missed by the user (especially if you opt in to using a menu type of action trip, resulting in only a "three-dots"-icon).

What we essentially  is after is something like you have at Azure devops:
![image](https://user-images.githubusercontent.com/12936286/143221050-cb460a04-9767-4dbe-a6e9-b195796fe90e.png)

**Comment by dkamburov** [comment]
You can use the context property of the action strip and pass a specific template to it. In this case it should be the cell of the grid. The context can be passed to the `show` API method of the action strip. It has its specifics due to the fact that it is designed mainly for row context of the grid, but if you can set `element` property with a `nativeElement` in it, the action strip will be rendered inside that element:
`actionstrip2.show({ element: { nativeElement: elem } })`

And to make it more clear here's the complete code sample:
```
    <igx-grid #grid2 [data]="data">
        <igx-column [field]="'ID'" [width]="'50px'">
            <ng-template igxCell let-cell="cell" let-val>
                <div #elem class="cell-template" (mouseover)="actionstrip2.show({ element: { nativeElement: elem } })">
                    
                </div>
            </ng-template>
        </igx-column>
    </igx-grid>

    <igx-action-strip #actionstrip2 [hidden]="true">
        <ng-container>
                <igx-icon *igxActionStripMenuItem (click)="doSomeAction()">alarm</igx-icon>
        </ng-container>
    </igx-action-strip>
```

Note that if put inside the grid template the action-strip will use the row as a context.

**Reply by Eralmidia** [reply]
@dkamburov Thanks, that helped a lot! One issue however: the strip will now only trigger when hovering the column where the strip is to be show. If I hover another column, it does of course not update, since the mouse over is only on one column:
![image](https://user-images.githubusercontent.com/12936286/144212002-155d11fe-1501-4ad0-99a9-ba89b30599ac.png)
 
 Is there an easy way to get the element ref to the first column in the second column in order to keep the strip in the same position no matter which column you are hovering?

**Reply by Eralmidia** [reply]
There is also another issue: Since the strip is outside the grid in your example, it will no longer automatically hide. I tried adding a mouseout in the igx-grid itself, with actionstrip.hide, and that does hide the strip when mouing the mouse outside the grid. The issue however is that since the strip is outside the grid, moving the mouse over one of the buttons in the strip also triggers the mouseout on the grid and hides the strip, rending the strip completely useless.

If I move the strip inside the hiding seems to work better, and I can actually click on the strip, but when inside the grid the grid want to put it to the far right in the row, and the grid ends up being confused whether the strip should be positioned in column one or at the default location at the right hand side of the grid.

**Reply by Eralmidia** [reply]
Managed to solve this one by using mouseleave on a div encapsulating both the grid and the strip instead.

**Reply by dkamburov** [reply]
Awesome, glad to hear you got this solved.

**Reply by Eralmidia** [reply]
Yes, but only the second issue tho. It will still only trigger when I hover over the column where the strip is shown. I would like for the strip to show if I hover over any column in the row, but that would require me to receive the element property in some other way.

**Reply by dkamburov** [reply]
I see, I though it was supposed to show only when you are on a specific cell.
How about adding mouseover to each cell and then using a specific one as a context of the action strip:

```html
   <igx-grid #grid2 [data]="data">
        <igx-column *ngFor="let c of columns" [sortable]="true" [field]="c.field" [header]="c.field"
            [width]="c.width" [pinned]='c.pinned' [hidden]='c.hidden'>
                <ng-template igxCell let-cell="cell" let-val>
                    <div class="cell-template" (mouseover)="showForSpecificCell(actionstrip2, grid2, cell);">
                    </div>
                </ng-template>
        </igx-column>

    </igx-grid>

    <igx-action-strip #actionstrip2 [hidden]="true">
        <ng-container>
                <igx-icon *igxActionStripMenuItem (click)="doSomeAction()">alarm</igx-icon>
        </ng-container>
    </igx-action-strip>
```

```typescript
    public showForSpecificCell(actionStrip, grid, cell, cellIndex = 0) {
        const row: IgxGridRowComponent = grid.gridAPI.get_row_by_index(cell.row.index);
        actionStrip.show({ element: { nativeElement: row.cells.toArray()[cellIndex].nativeElement } });
    }
```
